README.md

```diff
@@ -1,9 +1,65 @@
+
+### How does OpenPubkey ensure the nonce claim functions as nonce?
+
+In the user-identity scenario the CIC that contains the user's public key is hashed to the `nonce` claim in the ID Token. As OIDC requires that this field never repeat, OpenPubkey includes a random value, rz, in the CIC. Thus the hash of the CIC is always different and random. This maintains the required properties needed by the `nonce` claim in OIDC.
+
+```
+rz = crypto.random()
+CIC = {'rz': crypto.random(), 'upk': <publickey>, 'alg': 'EC256'}
+IDToken.nonce = SHA3(CIC)
+```
+
+### Does OpenPubkey present a privacy leak?
+
+The PK Tokens used by OpenPubkey contain the claims from the OIDC ID Tokens of the signer, so making them public, necessarily makes those claims public too. This may include elements of the signer’s identity such as the signer’s name or email addresses. It is up to users of OpenPubkey as to whether or not PK Tokens are made public.
+
+In a public artifact signing scenario, it could be argued that these very claims are the very claims upon which trust in the artifact should be based. However, some OIDC providers may include claims that the signer may wish to keep private. Users of OpenPubkey should consider carefully which OIDC providers to integrate with.
+
+Most OpenID Providers (OP) allow you to scope the fields in the ID Tokens. For instance Google’s OP is by default scoped to only include userinfo-email claims: name, email address and icon. Given that purpose of OpenPubkey is to enable parties to verify that a particular identity e.g., ethan@bastion.com, produced a particular signature, if you do not want signatures to be associated with OIDC identities, then OpenPubkey may not be a good fit for your use case.
+
+In other OpenPubkey deployment scenarios, such as those employed by BastionZero, the ID Tokens are not made publicly available.
+
+### Can the ID Tokens in Pubkeys be replayed against OIDC Resource Providers?
+
+Although not present in the original OpenPubkey paper, GQ signatures have now been integrated so that the OpenID Provider's signature can be stripped from the ID Token, and a proof of that signature published in its place. This prevents the ID Token present in the PK Token from being used against any OIDC resource providers as the original signature has been stripped. However the GQ Signature provides the same assurance that the signature provided.
```

Comment:
> However the GQ Signature provides the same assurance that the signature provided.

It might be good to further clarify to what signature the second mention of signature in the above sentence refers.